packages/0x.js/src/contract_wrappers/exchange_wrapper.ts

```diff
@@ -873,6 +876,19 @@ export class ExchangeWrapper extends ContractWrapper {
             throw new Error(errMessage);
         }
     }
+    /**
+     * Gets the latest OrderState of a signedOrder
+     * @param   signedOrder   The signedOrder
+     */
+    public async getOrderStateAsync(signedOrder: SignedOrder): Promise<OrderState> {
+        const balanceAndProxyAllowanceLazyStore = new BalanceAndProxyAllowanceLazyStore(
+            this._tokenWrapper,
+            BlockParamLiteral.Latest,
+        );
+        const orderFilledCancelledLazyStore = new OrderFilledCancelledLazyStore(this);
+        const orderStateUtils = new OrderStateUtils(balanceAndProxyAllowanceLazyStore, orderFilledCancelledLazyStore);
```

Comment:
It feels a bit hacky to use the LazyStore's here... We don't want any caching for this method.

The proper refactor would be passing a class that implements the abstract classes `BalanceAndAllowanceFetcher` and `OrderFilledCancelledFetcher` into `OrderStateUtils`, where their respective interfaces would be:

**balanceAndAllowanceFetcher**
```typescript
abstract class BalanceAndAllowanceFetcher {
  public abstract async getBalanceAsync(tokenAddress: string, userAddress: string): Promise<BigNumber>;
  public abstract async getProxyAllowanceAsync(tokenAddress: string, userAddress: string): Promise<BigNumber>;
}
``` 

**orderFilledCancelledFetcher**
```typescript
abstract class OrderFilledCancelledFetcher {
  public abstract async getFilledTakerAmountAsync(orderHash: string): Promise<BigNumber>;
  public abstract async getCancelledTakerAmountAsync(orderHash: string): Promise<BigNumber>;
}
```

The OrderWatcher can pass in the lazyStores, and ExchangeWrapper can pass in classes that simply fetch the latest information without caching. The OrderStateUtils doesn't care about the implementation, only that it receives the correct value.

Can you make this refactor happen? Once this is done, we can also instantiate these two fetchers once at ExchangeWrapper instantiation, rather then on every call to `getOrderStateAsync`?